File: Models/DTOs/CreateStudentDTO.cs
using DiplomaThesisDigitalization.Models.Entities;

namespace DiplomaThesisDigitalization.Models.DTOs
{
    public class CreateStudentDTO
    {
        public string Name { get; set; }
        public string Surname { get; set; }
        public DateTime DOB { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public string Gender { get; set; }
        public string Phone { get; set; }
        public string Address { get; set; }
        public int ECTS { get; set; }
        public string DegreeLevel { get; set; }
        public int FieldId { get; set; }  // Specializimi ~~ Field
        //public Field Field { get; set; }
        public int DepartmentId { get; set; }
        //public Department Department { get; set; }
    }
}

File: my-ts-react-app/src/app/models/DTOS/CreateStudentDTO.ts
import { Department } from "../Department";
import { Field } from "../Field";

export interface CreateStudentDTO {
  name: string;
  surname: string;
  dob: Date;
  email: string;
  password: string;
  gender: string;
  phone: string;
  address: string;
  ects: number;
  degreeLevel: string;
  fieldId: number;
  field: Field;
  departmentId: number;
  department: Department;
}
